engine/src/main/scala/cromwell/services/KeyValueServiceActor.scala

```diff
@@ -0,0 +1,72 @@
+package cromwell.services
+
+import akka.actor.{Actor, ActorRef, Props}
+import com.typesafe.config.Config
+import cromwell.backend.BackendJobDescriptor
+import cromwell.engine.db.DataAccess
+import cromwell.services.KeyValueServiceActor._
+import cromwell.services.ServiceRegistryActor.ServiceRegistryMessage
+
+import scala.concurrent.Future
+import scala.util.{Failure, Success}
+
+object KeyValueServiceActor {
+  trait KvMessage
+  trait KvAction extends KvMessage with ServiceRegistryMessage {
+    def serviceName = "KeyValue"
+  }
+  trait KvResponse extends KvMessage
+
+  case class KvPut(jobDescriptor: BackendJobDescriptor, key: String, value: Option[String]) extends KvAction
+  case class KvGet(jobDescriptor: BackendJobDescriptor, key: String) extends KvAction
+
+  case class KvPair(jobDescriptor: BackendJobDescriptor, key: String, value: Option[String]) extends KvResponse
```

Comment:
You're not missing anything.  The reason why I wrote it the way I did initially is that I perceived that as the simplest approach given the constraints I had to work with.  That's all.

I've already pushed the new design... I couldn't help but be reminded of the quote "a camel is a horse designed by a committee" :)